### ISSUE TITLE
Minicap install fails

### ISSUE THREAD

**Issue by HughJeffner** [HughJeffner/FFRK-LabMem#154]
Nox (debloated) > Samsung Galaxy S9+
Android 5

![image](https://user-images.githubusercontent.com/17887799/151253555-369c47b9-211a-4728-bda1-8df7948bd378.png)

**Comment by HughJeffner** [comment]
Looks like it is actually not liking something on the windows file system

Can you make sure adb is connected to the device then run the following command in the command line (in the bot directory):

`adb shell "getprop ro.product.cpu.abi | tr -d '\r'"`

**Comment by slartifer** [comment]
![image](https://user-images.githubusercontent.com/17887799/151258087-b8efc09d-9d45-4456-ae40-0f24067d7321.png)

**Comment by HughJeffner** [comment]
ah the `tr` command is not on that device.  I should just trim it after I get the result back instead

**Comment by slartifer** [comment]
Yeah options list offers
![image](https://user-images.githubusercontent.com/17887799/151258532-c25f4000-8a7a-4824-8f94-1e6afb535b69.png)

**Comment by HughJeffner** [comment]
Just to make sure see what the output of, `adb shell "getprop ro.product.cpu.abi"` is

**Comment by slartifer** [comment]
![image](https://user-images.githubusercontent.com/17887799/151258641-6209e41a-59bf-4548-803d-ceb6423a34f0.png)

**Comment by HughJeffner** [comment]
Gah

Well, this will be fixed in 7.2

**Comment by slartifer** [comment]
In the meantime, would it be bad if I ran it without the tr?

**Comment by HughJeffner** [comment]
You can manually push these 2 files with adb push command:

{abi} is `x86`
{apiLevel} is `21` or `22` depending on if you are on android 5.0 or 5.1
left hand path is windows file system, right hand side is android

`.\minicap\{abi}\bin\minicap` to `/data/local/tmp/minicap` 
`.\minicap\{abi}\lib\android-{apiLevel}/minicap.so` to `/data/local/tmp/minicap.so`

**Comment by HughJeffner** [comment]
Whoops forgot to set the permissions of those two files

Need to adb shell chmod 777 those 2 files

**Comment by slartifer** [comment]
That worked!  Well, sort of: LabMem now doesn't seem to recognize any buttons (at least not check inventory for auto start, or the OK button on a portal confirm dialogue).

![image](https://user-images.githubusercontent.com/17887799/151266779-8d780f4e-771d-4093-88bf-1aa6c6d0cc07.png)

**Comment by HughJeffner** [comment]
The only clue I have is your frame capture delay is 30-40ms where as mine is typically 80-90ms, dipping down to 40-50ms only when the screen is mostly black.  I suspect something with minicap isn't playing well with Nox and it is returning a blank screen or something.

To test this we can run minicap with the `-s` switch to do a single capture to a file:

`adb shell LD_LIBRARY_PATH=/data/local/tmp/ /data/local/tmp/minicap -P 1080x1920@1080x1920/0 -s ">/data/local/tmp/minicap.jpg"`

And then pull it to the bot directory:

`adb pull /data/local/tmp/minicap.jpg`

**Comment by slartifer** [comment]
Output from first command:

![image](https://user-images.githubusercontent.com/17887799/151293679-57ff1014-4edf-441a-bae3-49e453cc4e1d.png)

And resulting screencap:

![image](https://user-images.githubusercontent.com/17887799/151293699-00380cd9-87ac-452e-9253-7bfe7dab8da4.png)

Yup, it's getting blanks.  Alas.

**Comment by HughJeffner** [comment]
Well, I guess it was kinda good the install fails on Nox since that would cause a lot of confusion.  

The only thing I could do at this point is fork minicap and see if I can get it to work with Nox which would probably be a ton of work.  Thankfully, from what I can tell, the number of users running Nox is small.

Fixing the install causes another problem since I would have to check if getting a blank screen when verifying minicap is installed.  I'll at least do this and put a warning on screen the device is incompatible and auto-switch to Adb.

**Comment by slartifer** [comment]
Yup, it definitely sounds like we should just stop using Nox.  Ah well!
